### ISSUE TITLE
Use dataloader with @ResolveReference

### ISSUE THREAD

**Issue by ydomenjoud** [Adrinalin4ik/Nestjs-Graphql-Tools#27]
First of all, specials thanks for this awesome library.

**Describe the bug**
With Graphql Federation, you use @ResolveReference to resolve a reference to an entity.
Unfortunately,  @GraphqlLoader() and @Loader() annotations doesn't work with @ResolveReference.
I ran into this error : 
```ts
error {
  message: "Cannot read properties of undefined (reading 'path')",
  locations: [ { line: 1, column: 34 } ],
  path: [ '_entities', 0 ],
  extensions: {
    code: 'INTERNAL_SERVER_ERROR',
    stacktrace: [
      "TypeError: Cannot read properties of undefined (reading 'path')",
      '    at UsersResolver.descriptor.value (/work/users-application/node_modules/nestjs-graphql-tools/lib/loader.ts:125:51)',
      '    at /work/users-application/node_modules/@nestjs/core/helpers/external-context-creator.js:67:33',
      '    at processTicksAndRejections (node:internal/process/task_queues:96:5)',
      '    at async target (/work/users-application/node_modules/@nestjs/core/helpers/external-context-creator.js:74:28)'
    ]
  }
}
```

It seems to be related to line 96 in loader.ts , where ctx.getArgs() have different shape with graphql federation : 
```ts
  const args = ctx.getArgs();
  const { req } = args[2];
  const info = args[3];
```
On federated graph, req is in args[1] and info in args[2] .
Theses updates solved my issue : 
 ```ts
export const Loader = createParamDecorator((_data: unknown, ctx: ExecutionContext) => {
    const args = ctx.getArgs();
    let req;
    let info;
    if(args.length < 4) {
        req = args[1].req;
        info = args[2]
    } else {
        req = args[2].req;
        info = args[3]
    }
return {
....
```

**To Reproduce**
* clone https://github.com/nestjs/nest/tree/master/sample/31-graphql-federation-code-first ( download directory only with https://download-directory.github.io/ )
* install `npm i nestjs-graphql-tools` in `users-application`
* annotate method resolveReference of `users-application/src/users/users.resolver.ts`
```ts
  @ResolveReference()
  @GraphqlLoader()
  async resolveReference(
      @Loader() loader: LoaderData<User, number>,
      @Parent() reference: { __typename: string; id: number },
  ) {
    const ids = loader.ids;
    const users = this.usersService.findByIds(ids);
    return loader.helpers.mapManyToOneRelation(users, loader.ids, 'id')
  }
```
* add method findByIds to `users-application/src/users/users.service.ts` : 
```ts
  findByIds(idsList: number[]): User[] {
    return this.users.filter((user) => idsList.some(id => Number(id) === user.id));
  }
```
* install dependencies of 3 projects : `npm ci` in gateway,  posts-application,  users-application
* run all projects in order : 
  * `cd users-application && npm run start`
  * `cd posts-application && npm run start`
  * `cd gateway && npm run start`
* send graphql request to gateway
```graphql
{
  posts {
    id
    title
    authorId
    user {
      id
      name
    }
  }
}
```

**Expected behavior**
We expect no error and result return from the gateway

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 18
 - Browser chrome
 - Version 116.0.5845.179

**Additional context**
we can use also fieldName to search if this is a FederatedGraph by looking info.fieldName='_entities' too, or testing info.path.prev === undefined

**Comment by Adrinalin4ik** [comment]
@ydomenjoud thank you for such a good description. I'll try to sort it out a bit later and make you know once it's done.

**Comment by Adrinalin4ik** [comment]
@ydomenjoud this issue was fixed as part of 0.8.1 version. Please consider to upgrade to the latest version 0.8.2, it also include vulns fixes.

**Comment by Adrinalin4ik** [comment]
@ydomenjoud thank you for the description once again. I've added new section to readme related to federation and used your description there.

**Comment by Adrinalin4ik** [comment]
This issue was closed as it was fixed as part of 0.8.1 version.
